### PR TITLE
Fix image_profile resource read issue

### LINKED ISSUE #234: Faile to create a vSphere Image Mapping
**vRA version**
Version 8.1.0.9583 (16633378) 
**Terraform version**
Terraform v0.12.29
**terraform-provider-vra plugin version**
+ provider.vra v0.2.4

**Describe the bug**
When attempting to create a single vSphere Image Mapping by "terraform apply -auto-approve" we get error:

**Error: unknown error (status 409): {resp:0xc0001225a0} 
  on main.tf line 13, in resource "vra_image_profile" "this":
  13: resource "vra_image_profile" "this" {**

Doing the "Terraform plan" doesn't return an error.

**To Reproduce**
1. Terraform config file (for updates, config file before applying the change and the change made)

`provider vra {
  url           = var.vra_url
  refresh_token = var.vra_refresh_token
  insecure = true
}

data "vra_region" "this" {
  cloud_account_id = "5ea2b589-dc80-49f9-8c67-66178a8b0f1f"
  region           = "Datacenter:datacenter-21"
}

# Create a new image profile
resource "vra_image_profile" "this" {
  name        = "vsphere-image"
  description = "vSphere image profile created by Terraform"
  region_id   = data.vra_region.this.id

  image_mapping {
    name       = "windows"
    image_name = "win2016bis"
  }
}
`
We collected the "cloud_account_id" from API and it seems aligned to the one seen at the URL when in vRA URL for Cloud Accounts
also "region" value is aligned to what the API call returns in "enabledRegionIds", "image_name" is a valid template displayed when adding the image directly from vRA UI.

2. Terraform command

`# terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.vra_region.this: Refreshing state...

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # vra_image_profile.this will be created
  + resource "vra_image_profile" "this" {
      + description = "vSphere image profile created by Terraform"
      + id          = (known after apply)
      + name        = "vsphere-image"
      + region_id   = "c6e082a3-cbe7-4c47-971b-de0c6d93531d"

      + image_mapping {
          + external_id        = (known after apply)
          + external_region_id = (known after apply)
          + image_name         = "win2016bis"
          + name               = "windows"
          + organization       = (known after apply)
          + os_family          = (known after apply)
          + owner              = (known after apply)
          + private            = (known after apply)
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

------------------------------------------------------------------------`

Then we try "terraform apply -auto-approve" we get the error

3. Error

`data.vra_region.this: Refreshing state...
vra_image_profile.this: Creating...

Error: unknown error (status 409): {resp:0xc0009dc1b0} 

  on main.tf line 13, in resource "vra_image_profile" "this":
  13: resource "vra_image_profile" "this" {
`
Looking at the vRA Provider logs, the API seems to be complaining that the image profile already exists ( which doesn't ), responding with error:

`2020-08-27T19:50:11.093-0400 [DEBUG] plugin.terraform-provider-vra_v0.2.4: {
2020-08-27T19:50:11.093-0400 [DEBUG] plugin.terraform-provider-vra_v0.2.4:  "message": "Image profile for the same region already exists",
2020-08-27T19:50:11.093-0400 [DEBUG] plugin.terraform-provider-vra_v0.2.4:  "statusCode": 409,
2020-08-27T19:50:11.093-0400 [DEBUG] plugin.terraform-provider-vra_v0.2.4:  "errorCode": 0,
2020-08-27T19:50:11.093-0400 [DEBUG] plugin.terraform-provider-vra_v0.2.4:  "serverErrorId": "3176c4fb-8d9a-4616-96d8-cc95860ace7c",
2020-08-27T19:50:11.093-0400 [DEBUG] plugin.terraform-provider-vra_v0.2.4:  "documentKind": "com:vmware:xenon:common:ServiceErrorResponse"
2020-08-27T19:50:11.093-0400 [DEBUG] plugin.terraform-provider-vra_v0.2.4: }`

**Expected behavior**
Create a vSphere based Image mapping.

**Screenshots**

**Logs**
Attach logs to help debug the issue
1. Attach vra-terraform.log
2. Attach Terraform console log (Enable terraform logs following the steps mentioned in https://www.terraform.io/docs/configuration/environment-variables.html)
3. crash.log(if any)

**Desktop (please complete the following information):**
 - CentOS Linux release 7.7.1908 (Core)

**Additional context**

### PR DESCRIPTION
**Pull request by dmettem**
* Fix image_profile resource read issue - allows the import to function correctly
* Add data source for image_profile
* Add example for image_profile resource and data source

Fixes #234.

Signed-off-by: Deepak Mettem <dmettem@vmware.com>

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -278,6 +278,8 @@ github.com/vmihailenco/msgpack v4.0.1+incompatible h1:RMF1enSPeKTlXrXdOcqjFUElyw
 github.com/vmihailenco/msgpack v4.0.1+incompatible/go.mod h1:fy3FlTQTDXWkZ7Bh6AcGMlsjHatGryHQYUTf1ShIgkk=
 github.com/vmware/vra-sdk-go v0.2.10 h1:nw1NSroINtw8lnaGUtTBuvDWEepBjEkTjEukYSqfJxM=
 github.com/vmware/vra-sdk-go v0.2.10/go.mod h1:W6OERth9ssa4FgvS98SOVusGgBNWJs+D4Hq/5ueTgIQ=
+github.com/vmware/vra-sdk-go v0.2.11 h1:QSnlKItvEmnhd9yfaCLesTlb/I928F/OhQ7LpaGljyI=
```
Comment (markpeek) [inline go.sum:281]: Seeing multiple versions in this file. Do we need to run `go mod tidy`?

File: vra/data_source_image_profile.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+func dataSourceImageProfile() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceImageProfileRead,
+
+		Schema: map[string]*schema.Schema{
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Date when the entity was created. The date is in ISO 8601 and UTC.",
```
Comment (markpeek) [inline vra/data_source_image_profile.go:21]: NIce to see Descriptions added!

File: vra/data_source_image_profile.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	apiClient := m.(*Client).apiClient
+	var imageProfile *models.ImageProfile
+	var filter string
+
+	id := d.Get("id").(string)
+	name := d.Get("name").(string)
+	configFilter := d.Get("filter").(string)
+	regionID := d.Get("region_id").(string)
+
+	if id == "" && name == "" && configFilter == "" && regionID == "" {
```
Comment (markpeek) [inline vra/data_source_image_profile.go:83]: Do we need to check that only one of these is set?

**Review by markpeek — APPROVED** [review]
A couple of comments but otherwise LGTM.

File: go.sum (hunk 1 above)
Comment (dmettem) [reply go.sum:279]: Done.

File: vra/data_source_image_profile.go (hunk 3 above)
Comment (dmettem) [reply vra/data_source_image_profile.go:87]: Done.

**Review by markpeek — APPROVED** [review]
LGTM

**Pull request merged**